src/www/ui/search.php

```diff
@@ -141,9 +161,26 @@ function Output()
       $this->vars["Filename"] = $Filename;
     }
 
-    $tag = GetParm("tag", PARM_RAW);
-    if (! empty($tag)) {
-      $CriteriaCount ++;
+    $Upload = GetParm("upload",PARM_INTEGER);
+    $SelectedUploadName = "All uploads";
+    if ($Upload != 0) 
+    {
+      $CriteriaCount++;
+      $GETvars .= "&upload=" . urlencode($Upload);
+      $this->vars["Upload"] = $Upload;
+      foreach($uploadsArray as $row)
+      {
+        if($row['upload_pk'] == $Upload)
+        {
+          $SelectedUploadName = $row['upload_filename'] . " from " .  substr($row['upload_ts'],0,-10);
```

Comment:
How about using the `Convert2BrowserTime()` from #1667?

Reply:
Changed